promql/engine.go

```diff
@@ -379,12 +394,12 @@ func (ng *Engine) SetQueryLogger(l QueryLogger) {
 }
 
 // NewInstantQuery returns an evaluation query for the given expression at the given time.
-func (ng *Engine) NewInstantQuery(q storage.Queryable, qs string, ts time.Time) (Query, error) {
+func (ng *Engine) NewInstantQuery(q storage.Queryable, opts QueryOpts, qs string, ts time.Time) (Query, error) {
```

Comment:
I really like this, because we want to jave later on per-query lookback delta, it could be a query opt as well.

Reply:
Can we have QueryOpts as a pointer ? we are passing that downstream mutliple times, creating un-needed allocs. it would also be nicer in tests.